app/controllers/comments_controller.rb

```diff
@@ -30,7 +30,13 @@ def create
 
   def destroy
     @comment = Comment.find(params[:id])
-    @comment&.destroy
+    if has_children?(@comment)
+      @comment.update_attribute(:hidden, true)
+      @partial = 'comments/hidden_comment'
+    else
+      @comment&.destroy_with_hidden_parents
+      @partial = 'shared/empty_partial'
+    end
```

Comment:
Teraz nie usuwamy komentarza, gdy ma dzieci, tylko go 'chowamy'.

Jednak jeśli ktoś potem usunie jego dzieci, to rodzic by został taki schowany bez celu. Dlatego coś musi go też usunąć w tej sytuacji. Po to jest .destroy_with_hidden_parents.

Nie byłem pewien, czy on powinien być w modelu, ale w kontrolerze nie da się zrobić rekurencji usuwania takiej.

Reply:
chyba pasuje w modelu. Ja nie wiem czy nazwa tej metody nie powinna jeszcze byc troche inna - cos co troche lepiej wyjasnia co ona robi, ale generalnie chyba ok